### ISSUE TITLE
Distribute foundation.lib for VS2017 and VS2019

### ISSUE THREAD

**Issue by niklas-ourmachinery** [OurMachinery/themachinery-public#194]
Visual Studio static libraries are not compatible between VS2017 and VS2019, so if we want the beta package to compile for both VS2017 and VS2019 we need to distribute VS2017 libraries to.

If you try to compile on VS2017 with the VS2019 library you get errors like this:

```
foundation.lib(job_system.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4 [C:\Users\Jason\Desktop\machinery_beta_2020.3\samples\build\simple_3d_exe\simple-3d-exe.vcxproj]
```

Reference: https://forums.ourmachinery.com/t/no-premake5-library-setup/552/5

**Comment by marynate** [comment]
Hi, bump to this page while trying to find out how to feed a vs2019 library to build with vs2017 without trigger the "__CxxFrameHandler4" linker error for my own project.
I solved the issue with get rid of "/EHsc" compiler flag and adding "_HAS_EXCEPTIONS=0" predefine.
(If you do need exception handling in your library, there's probably way to tell vs2019 to use old FH3 instead of FH4, but haven't try that myself)

**Comment by niklas-ourmachinery** [comment]
Fixed in 2020.5.
